### PR TITLE
bpo-45847: Port _posixshmem to PY_STDLIB_MOD (GH-29738)

### PR DESCRIPTION
**Pull request by erlend-aasland**
https://bugs.python.org/issue45847

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -6251,6 +6248,9 @@ PY_STDLIB_MOD([fcntl],
   [], [$FCNTL_LIBS])
 PY_STDLIB_MOD([mmap],
   [], [test "$ac_cv_header_sys_mman_h" = "yes" -a "$ac_cv_header_sys_stat_h" = "yes"])
+PY_STDLIB_MOD([_posixshmem],
+  [], [test "$have_posix_shmem" = "yes"],
```
Comment (erlend-aasland) [inline configure.ac:6252]: Would you prefer to have it this way, @tiran? I'm not sure what's most correct here.

```suggestion
  [test "$have_posix_shmem" = "yes"], [],
```

File: Modules/Setup.stdlib.in [hunk 2]
```diff
@@ -113,6 +113,7 @@
 @MODULE_NIS_TRUE@nis nismodule.c
 # needs sys/soundcard.h or linux/soundcard.h (Linux, FreeBSD)
 @MODULE_OSSAUDIODEV_TRUE@ossaudiodev ossaudiodev.c
+@MODULE__POSIXSHMEM_TRUE@_posixshmem _multiprocessing/posixshmem.c
```
Comment (erlend-aasland) [inline Modules/Setup.stdlib.in]: ... or we could move this to a "multiprocessing" section

**Review by tiran — APPROVED** [review]

File: Modules/Setup.stdlib.in (hunk 2 above)
Comment (tiran) [reply Modules/Setup.stdlib.in]: Yeah. that makes more sense.

File: configure.ac (hunk 1 above)
Comment (tiran) [reply configure.ac:6252]: The first check is for feature flags like ``--with-builtin-hashlib-hashes`` or ``--disable-test-modules``

**Review by tiran — APPROVED** [review]

**Pull request merged**